web/pdf_page_view.js

```diff
@@ -590,16 +580,10 @@ class PDFPageView {
     }
     this.structTreeLayer?.hide();
 
-    if (!zoomLayerNode) {
-      if (this.canvas) {
-        this.#viewportMap.delete(this.canvas);
-        // Zeroing the width and height causes Firefox to release graphics
-        // resources immediately, which can greatly reduce memory consumption.
-        this.canvas.width = 0;
-        this.canvas.height = 0;
-        delete this.canvas;
-      }
-      this._resetZoomLayer();
+    if (!keepCanvasWrapper && this.#canvasWrapper) {
+      this.#canvasWrapper.remove();
```

Comment:
Isn't the DOM element already removed above, courtesy of the following line?
https://github.com/mozilla/pdf.js/blob/f4b09e9a62343f820da9a0eeee1baafbdc8d4e8f/web/pdf_page_view.js#L557

If that's accurate, maybe just add a comment here about it already being removed?